### PR TITLE
[feature] Migrate event collection feature

### PR DESCRIPTION
**Pull request by khewonc**
### What does this PR do?

Migrate event collection feature

### Motivation

What inspired you to submit this pull request?

### Additional Notes

This uses code derived from https://github.com/DataDog/datadog-operator/pull/485 in `pkg/controller/utils/shared_utils.go` to retrieve the expected names for the leader election and event collection token resource names.

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: controllers/datadogagent/feature/eventcollection/feature.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+func buildEventCollectionFeature(options *feature.Options) feature.Feature {
+	eventCollectionFeat := &eventCollectionFeature{}
+
+	return eventCollectionFeat
+}
+
+type eventCollectionFeature struct {
+	nodeAgentEnable    bool
+	clusterAgentEnable bool
```
Comment (celenechang) [inline controllers/datadogagent/feature/eventcollection/feature.go:37]: are these params needed?

File: controllers/datadogagent/feature/eventcollection/feature.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+// ConfigureV1 use to configure the feature from a v1alpha1.DatadogAgent instance.
+func (f *eventCollectionFeature) ConfigureV1(dda *v1alpha1.DatadogAgent) (reqComp feature.RequiredComponents) {
+	f.owner = dda
+	config := dda.Spec.Agent.Config
+	dcaConfig := dda.Spec.ClusterAgent.Config
+
+	if *config.LeaderElection {
+		// cluster agent
+		if *dcaConfig.CollectEvents {
```
Comment (celenechang) [inline controllers/datadogagent/feature/eventcollection/feature.go:70]:
```suggestion
		if apiutils.BoolValue(dcaConfig.CollectEvents) {
```

File: controllers/datadogagent/feature/eventcollection/feature.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+			f.rbacSuffix = common.ClusterAgentSuffix
+
+			reqComp = feature.RequiredComponents{
+				ClusterAgent: feature.RequiredComponent{IsRequired: apiutils.NewBoolPointer(true)},
+			}
+			return reqComp
+		}
+
+		// node agent
+		if *config.CollectEvents {
```
Comment (celenechang) [inline controllers/datadogagent/feature/eventcollection/feature.go:82]:
```suggestion
		if apiutils.BoolValue(config.CollectEvents) {
```

File: apis/datadoghq/v1alpha1/utils.go [hunk 4]
```diff
@@ -27,6 +27,13 @@ func GetConfName(owner metav1.Object, conf *CustomConfigSpec, defaultName string
 	return fmt.Sprintf("%s-%s", owner.GetName(), defaultName)
 }
 
+// GetAgentServiceAccount returns the node agent serviceAccountName
+func GetAgentServiceAccount(dda *DatadogAgent) string {
```
Comment (celenechang) [inline apis/datadoghq/v1alpha1/utils.go:31]: i noticed there is a similar method https://github.com/DataDog/datadog-operator/blob/c07038fbf0b8993798ec5693cf6f191d7c369feb/controllers/datadogagent/utils.go#L1833

(that file also seems to have duplicates of `GetClusterAgentServiceAccount` and `GetClusterChecksRunnerServiceaccount`)

Comment (khewonc) [reply apis/datadoghq/v1alpha1/utils.go:31]: I tried using the controller utils and that caused a circular import error with go, probably from [importing eventcollection](https://github.com/DataDog/datadog-operator/pull/500/files#diff-827f2f59a3c51c89207aae6aeab23a903d8451f0f9123064bd71582b14e527abR38) in the `controller.go` file. I left it as is for now, but would be open to moving around those utility functions if that would be better.

**Review by celenechang — APPROVED** [review]

**Pull request merged**